internal/internal_worker.go

```diff
@@ -583,6 +583,8 @@ func (r *registry) RegisterWorkflowWithOptions(
 		if strings.HasPrefix(options.Name, temporalPrefix) {
 			panic(temporalPrefixError)
 		}
+		r.Lock()
+		defer r.Unlock()
```

Comment:
Not a big deal, but can probably just move the lock/unlock from several lines down to the top of this function so it covers future code too